src/webgpu/api/operation/render_pipeline/culling_tests.spec.ts

```diff
@@ -38,30 +38,106 @@ function faceColor(face: 'cw' | 'ccw', frontFace: GPUFrontFace, cullMode: GPUCul
   }
 }
 
+function checkCornerPixels(
+  t: GPUTest,
+  texture: GPUTexture,
+  format: SizedTextureFormat,
+  expectedTopLeftColor: Uint8Array,
+  expectedBottomRightColor: Uint8Array
+) {
+  t.expectSinglePixelIn2DTexture(texture, format, { x: 0, y: 0 }, { exp: expectedTopLeftColor });
+  t.expectSinglePixelIn2DTexture(
+    texture,
+    format,
+    { x: texture.width - 1, y: texture.height - 1 },
+    { exp: expectedBottomRightColor }
+  );
+}
+
+function drawFullClipSpaceTriangleAndCheckCornerPixels(
+  t: GPUTest,
+  texture: GPUTexture,
+  format: SizedTextureFormat,
+  topology: GPUPrimitiveTopology,
+  color: Uint8Array,
+  depthStencil: GPUDepthStencilState,
+  depthStencilAttachment: GPURenderPassDepthStencilAttachment,
+  expectedTopLeftColor: Uint8Array,
+  expectedBottomRightColor: Uint8Array
+) {
+  const { device } = t;
+  const encoder = device.createCommandEncoder();
+  const pass = encoder.beginRenderPass({
+    colorAttachments: [
+      {
+        view: texture.createView(),
+        loadOp: 'load',
+        storeOp: 'store',
+      },
+    ],
+    depthStencilAttachment,
+  });
+
+  pass.setPipeline(
+    device.createRenderPipeline({
+      layout: 'auto',
+      vertex: {
+        module: device.createShaderModule({
+          code: `
+            @vertex fn main(
+              @builtin(vertex_index) VertexIndex : u32
+              ) -> @builtin(position) vec4<f32> {
+                var pos : array<vec2<f32>, 3> = array<vec2<f32>, 3>(
+                  vec2<f32>(-1.0, -1.0),
+                  vec2<f32>( 3.0, -1.0),
+                  vec2<f32>(-1.0,  3.0));
+              return vec4<f32>(pos[VertexIndex], 0.0, 1.0);
+            }`,
+        }),
+        entryPoint: 'main',
+      },
+      fragment: {
+        module: device.createShaderModule({
+          code: `
+            @fragment fn main() -> @location(0) vec4<f32> {
+              return vec4f(${Array.from(color).map(v => v / 255)});
+            }`,
+        }),
+        entryPoint: 'main',
+        targets: [{ format }],
+      },
+      primitive: {
+        topology,
+      },
+      depthStencil,
+    })
+  );
+  pass.draw(3);
+  pass.end();
+
+  device.queue.submit([encoder.finish()]);
+
+  checkCornerPixels(t, texture, format, expectedTopLeftColor, expectedBottomRightColor);
+}
+
 export const g = makeTestGroup(GPUTest);
 
 g.test('culling')
```

Comment:
Ideally while we're here and you've been looking at the test code, we should write an actual description for this test

Reply:
done